js/core/src/init.ts

```diff
@@ -290,19 +292,24 @@ export function create_components(initial_layout: ComponentMeta | undefined): {
 			);
 		}
 
-		if (instance.type === "tabs") {
-			instance.children =
-				instance?.children?.map((c) => ({
-					...c,
-					props: {
-						...c.props,
-						id: c.props.id || c.id
-					}
-				})) || [];
-			const child_tab_items = instance.children?.filter(
+		if (instance.type === "tabs" && !instance.props.initial_tabs) {
```

Comment:
@hannahblair This is the main fix, we want to avoid blowing away the tabs but we need to set initial tabs for the first render. This way we will set them once but never again which prevent us overwriting them unnessecarily.